feat: add code-action quickfix

kevineor opened the pull request:
draft for https://github.com/hashicorp/terraform-ls/issues/1530

remilapeyre commented on internal/decoder/validations/missing_required_attribute.go:
```diff
@@ -47,9 +47,14 @@ func (mra MissingRequiredAttribute) Visit(ctx context.Context, node hclsyntax.No
-						Detail:   fmt.Sprintf("An attribute named %q is required here", name),
+						Detail:   fmt.Sprintf("Description : %q", attr.Description.Value),
 						Subject:  nodeType.SrcRange.Ptr(),
+						Extra: DiagnosticExtraMissingRequiredAttribute{
```
```suggestion
						Extra: map[string]interface{}{
```

remilapeyre commented on internal/langserver/handlers/code_action.go:
```diff
@@ -74,6 +84,50 @@ func (svc *service) textDocumentCodeAction(ctx context.Context, params lsp.CodeA
+			var extractedMissingAttrs []DiagnosticExtraMissingRequiredAttribute
+
+			for _, diag := range params.Context.Diagnostics {
+				var newMissingAttr DiagnosticExtraMissingRequiredAttribute
```
```
missingAttributeRaw, found := diag.Data["MissingAttribute"]
if !found {
    continue
}
missingAttribute, ok := missingAttributeRaw.(string)
if !ok {
    // For some reason MissingAttribute does not have the expected type, 
    // we continue checking the rest of the diagnostics.
    // Maybe log an error?
    continue
}
extractedMissingAttrs = append(extractedMissingAttrs, missingAttribute)
```

eltociear committed:
docs: update syntax-highlighting.md (#1528)

occuring -> occurring

jpogran committed:
Alert on wsl.localhost Path (#1522)

This adds path detection for network WSL paths to warn the user to open the path in WSL instead.

A user can open a WSL path directly using the network mount path, which results in opening inside the host VS Code instance instead of inside the WSL instance. This is a problem for terraform-ls, which can't resolve the network paths correctly.

dependabot[bot] committed:
build(deps): Bump github.com/hashicorp/terraform-json (#1541)

Bumps [github.com/hashicorp/terraform-json](https://github.com/hashicorp/terraform-json) from 0.18.0 to 0.19.0.
- [Release notes](https://github.com/hashicorp/terraform-json/releases)
- [Commits](https://github.com/hashicorp/terraform-json/compare/v0.18.0...v0.19.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-json
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): Bump github.com/vektra/mockery/v2 from 2.38.0 to 2.39.0 (#1542)

Bumps [github.com/vektra/mockery/v2](https://github.com/vektra/mockery) from 2.38.0 to 2.39.0.
- [Release notes](https://github.com/vektra/mockery/releases)
- [Changelog](https://github.com/vektra/mockery/blob/master/docs/changelog.md)
- [Commits](https://github.com/vektra/mockery/compare/v2.38.0...v2.39.0)

---
updated-dependencies:
- dependency-name: github.com/vektra/mockery/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): Bump github.com/hashicorp/terraform-exec (#1543)

Bumps [github.com/hashicorp/terraform-exec](https://github.com/hashicorp/terraform-exec) from 0.19.0 to 0.20.0.
- [Release notes](https://github.com/hashicorp/terraform-exec/releases)
- [Changelog](https://github.com/hashicorp/terraform-exec/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-exec/compare/v0.19.0...v0.20.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-exec
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): Bump github.com/vektra/mockery/v2 from 2.39.0 to 2.39.1 (#1544)

Bumps [github.com/vektra/mockery/v2](https://github.com/vektra/mockery) from 2.39.0 to 2.39.1.
- [Release notes](https://github.com/vektra/mockery/releases)
- [Changelog](https://github.com/vektra/mockery/blob/master/docs/changelog.md)
- [Commits](https://github.com/vektra/mockery/compare/v2.39.0...v2.39.1)

---
updated-dependencies:
- dependency-name: github.com/vektra/mockery/v2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jpogran committed:
Rename TokenTypeCapsule to TokenTypeComplex (#1529)

* Rename TokenTypeCapsule to TokenTypeComplex

This renames the existing token type TokenTypeCapsule to TokenTypeComplex to aid readability and reflect existing naming conventions.

This token represents `list` in `type = list(string)` and similar, i.e. the name of the complex type. The term "capsule" has slightly different meaning in cty and HCL and we should not use it in this context.

* Update to hcl-lang@latest

* Update docs

dependabot[bot] committed:
build(deps): Bump github.com/hashicorp/terraform-json (#1547)

Bumps [github.com/hashicorp/terraform-json](https://github.com/hashicorp/terraform-json) from 0.19.0 to 0.20.0.
- [Release notes](https://github.com/hashicorp/terraform-json/releases)
- [Commits](https://github.com/hashicorp/terraform-json/compare/v0.19.0...v0.20.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-json
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kevineor committed:
feat: add code-action quickfix

fix: do not use struct for missing attributes

kevineor committed:
fix: add quickfix to expected codeActionKinds
